test/services/kubernetes_service_test.rb

```diff
@@ -67,6 +82,75 @@ def after_teardown
     service.call(proxy)
   end
 
+  test 'create native ingress when not in openshift' do
+    ENV['KUBERNETES_NAMESPACE'] = 'zync'
+    ENV['KUBE_TOKEN'] = strict_encode64('token')
+    ENV['KUBE_SERVER'] = 'http://localhost'
+    ENV['KUBE_CA'] = encode64 <<~CERTIFICATE
+      -----BEGIN CERTIFICATE-----
+      MIIBZjCCAQ2gAwIBAgIQBHMSmrmlj2QTqgFRa+HP3DAKBggqhkjOPQQDAjASMRAw
+      DgYDVQQDEwdyb290LWNhMB4XDTE5MDQwNDExMzI1OVoXDTI5MDQwMTExMzI1OVow
+      EjEQMA4GA1UEAxMHcm9vdC1jYTBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABGG2
+      NDgiBuXNVWVVxrDNVjPsKm14wg76w4830Zn3K24u03LJthzsB3RPJN9l+kM7ryjg
+      dCenDYANVabMMQEy2iGjRTBDMA4GA1UdDwEB/wQEAwIBBjASBgNVHRMBAf8ECDAG
+      AQH/AgEBMB0GA1UdDgQWBBRfJt1t0sAlUMBwfeTWVv2v4XNcNjAKBggqhkjOPQQD
+      AgNHADBEAiB+MlaTocrG33AiOE8TrH4N2gVrDBo2fAyJ1qDmjxhWvAIgPOoAoWQ9
+      qwUVj52L6/Ptj0Tn4Mt6u+bdVr6jEXkZ8f0=
+      -----END CERTIFICATE-----
+    CERTIFICATE
```

Comment:
@akostadinov You are correct, they are the same and I understand your point. However, I am not really familiar with writing tests in Ruby. Is something like this sufficient?

`class Integration::KubernetesServiceTest < ActiveSupport::TestCase
  include Base64

  def before_setup
    @_env = ENV.to_hash
    super
  end

  def after_teardown
    ENV.replace(@_env)
    super
  end

  before do
    ENV['KUBERNETES_NAMESPACE'] = 'zync'
    ENV['KUBE_TOKEN'] = strict_encode64('token')
    ENV['KUBE_SERVER'] = 'http://localhost'
    ENV['KUBE_CA'] = encode64 <<~CERTIFICATE
      -----BEGIN CERTIFICATE-----
      MIIBZjCCAQ2gAwIBAgIQBHMSmrmlj2QTqgFRa+HP3DAKBggqhkjOPQQDAjASMRAw
      DgYDVQQDEwdyb290LWNhMB4XDTE5MDQwNDExMzI1OVoXDTI5MDQwMTExMzI1OVow
      EjEQMA4GA1UEAxMHcm9vdC1jYTBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABGG2
      NDgiBuXNVWVVxrDNVjPsKm14wg76w4830Zn3K24u03LJthzsB3RPJN9l+kM7ryjg
      dCenDYANVabMMQEy2iGjRTBDMA4GA1UdDwEB/wQEAwIBBjASBgNVHRMBAf8ECDAG
      AQH/AgEBMB0GA1UdDgQWBBRfJt1t0sAlUMBwfeTWVv2v4XNcNjAKBggqhkjOPQQD
      AgNHADBEAiB+MlaTocrG33AiOE8TrH4N2gVrDBo2fAyJ1qDmjxhWvAIgPOoAoWQ9
      qwUVj52L6/Ptj0Tn4Mt6u+bdVr6jEXkZ8f0=
      -----END CERTIFICATE-----
    CERTIFICATE
  end
  ...
`

It's a bit unclear to me how `before do` works in relation to say `def before_setup`. And which would be more appropriate here.

Reply:
Should work. Also changing env in `before_setup` should also be fine. It will do once for all tests in the class. But make sure to put it after `@_env = ENV.to_hash`.

Reply:
It's been so long and I need to rebase from master anyways at this point. It looks like that particular block is in a `setup do` now. So I will do that now.

Reply:
Yea that merge didn't go the way I wanted it to. I think I am going to restart from master and just change the two files independently.